modules/mcuboot/Kconfig

```diff
@@ -61,6 +61,23 @@ config MCUBOOT_FLASH_WRITE_BLOCK_SIZE
 
 endif # BOOTLOADER_MCUBOOT
 
+config BOOT_IMAGE_ACCESS_HOOK_CUSTOM
+	bool "Custom image access hooks"
+	depends on BOOT_IMAGE_ACCESS_HOOKS
+	help
+	  Enabling this option allows you to use a custom hooks file for the
+	  MCUBoot image access hooks. The same option needs to be set for the
+	  corresponding mcuboot child image.
+
+config BOOT_IMAGE_ACCESS_HOOK_CUSTOM_PATH
+	string "Path to hooks file"
+	default "nrf_custom_hooks.c"
```

Comment:
we should not refer to c files from Kconfig.

We have CMake for setting adding and controlling files.

If you want to avoid adding a C file from a repo, then we have a pattern for that.
Will post example later.

Reply:
As example, Zephyr provides an entropy driver for nRF:
https://github.com/zephyrproject-rtos/zephyr/blob/577c678df603798a32aad50918f7683e98eb7f0e/drivers/entropy/Kconfig.nrf5#L14
```
menuconfig ENTROPY_NRF5_RNG
```
and in CMake: https://github.com/zephyrproject-rtos/zephyr/blob/577c678df603798a32aad50918f7683e98eb7f0e/drivers/entropy/CMakeLists.txt#L13
```
zephyr_library_sources_ifdef(CONFIG_ENTROPY_NRF5_RNG           entropy_nrf5.c)
```

but where a downstream implementation can be provided and upstream can be forcefully disabled using `ENTROPY_NRF_FORCE_ALT`.
Ref: https://github.com/zephyrproject-rtos/zephyr/blob/577c678df603798a32aad50918f7683e98eb7f0e/drivers/entropy/Kconfig.nrf5#L7

and then downstream the alternative implementation can be amended to the proper Zephyr library:
https://github.com/nrfconnect/sdk-nrf/blob/6e95782dc3208f66ea621ef2eff427690ecbf253/drivers/entropy/CMakeLists.txt#L7